### PR TITLE
fix(systemtrayicon): #3419, don't activate qTox widget on tray icon click in Unity backend

### PR DESCRIPTION
**Pull request by abbat**
Seems Unity has own opinion about usability and user experience :)

---

This change is [Reviewable](https://reviewable.io/reviews/tux3/qtox/3423)

### CODE REVIEW

File: src/widget/widget.cpp [hunk 1]
```diff
@@ -1801,7 +1801,12 @@ void Widget::onTryCreateTrayIcon()
             trayMenu->addAction(actionQuit);
             icon->setContextMenu(trayMenu);
 
+            #ifdef ENABLE_SYSTRAY_UNITY_BACKEND
```
Comment (zetok) [inline src/widget/widget.cpp:1805]: perhaps a comment could be added that this works around issue #3419 ?

File: src/widget/systemtrayicon.h [hunk 2]
```diff
@@ -38,6 +38,7 @@ class SystemTrayIcon : public QObject
     void hide();
     void setVisible(bool);
     void setIcon(QIcon &icon);
+    SystrayBackendType backend() const { return backendType; }
```
Comment (sudden6) [inline src/widget/systemtrayicon.h:41]: Please move the implementation to the cpp file.

**Comment by sudden6** [comment]
@abbat see my comment, are you going to change this or should I do it myself?

**Comment by abbat** [comment]
@sudden6, Please, do it yourself - I already delete clone for this PR.

**Pull request closed without merging**